core/session.ts

```diff
@@ -347,14 +363,7 @@ export class Session {
     allActions.forEach(action => {
       const uniqueDependencies: { [dependency: string]: boolean } = {};
```

Comment:
this can be simplified now:

`action.dependencies = [...new Set(action.dependencies || [])];`

Reply:
ES6 is fantastic! Done.